### PR TITLE
Update ITK to v4.12

### PR DESCRIPTION
**Pull request by mathturtle**
@cjh1 At some point can you show me how to use data.kitware.com for the files?  For now I just put these on the tomviz.org site.

@cryos Binaries uploaded and ready to go.

### CODE REVIEW

**Comment by cjh1** [comment]
@mathturtle Sure

File: itk-binaries/osx/make_itk_bundle_osx.sh [hunk 1]
```diff
@@ -42,15 +46,16 @@ $cmake_path/cmake -DCMAKE_BUILD_TYPE:STRING=Release \
   -DBUILD_EXAMPLES:BOOL=OFF \
   -DBUILD_SHARED_LIBS:BOOL=ON \
   "-DCMAKE_INSTALL_PREFIX:PATH=$workdir/install" \
-  "-DNUMPY_INCLUDE_DIR:PATH=$tvsb_dir/build/install/lib/python3.6/site-packages/numpy/core/include" \
+  "-DNUMPY_INCLUDE_DIR:PATH=$tvsb_dir/build/install/lib/python2.7/site-packages/numpy/core/include" \
```
Comment (cryos) [inline itk-binaries/osx/make_itk_bundle_osx.sh]: This doesn't look right - we want to continue to target Python 3.6.

File: versions.cmake [hunk 2]
```diff
@@ -130,23 +130,22 @@ else()
...
-    URL "http://www.tomviz.org/files/itk-4.11.0-windows-64bit.zip"
-    URL_MD5 "8357ebd1ccce7d2366a96c569e2b3d11")
+    URL "http://www.tomviz.org/files/itk-4.12.0-windows-64bit.zip"
+    URL_HASH SHA512=${_itk_sha512})
   add_revision(fftw
     URL "http://www.tomviz.org/files/fftw-3.3.4-windows-64bit.zip"
     URL_MD5 "90ca2a2cd26c62bc85b11ec7f165e716")
 elseif(APPLE)
-  set(_itk_sha512 "00b2073761fb6d456bbaf8883a0362a56c2a49228441408e82d4293b1633072fbcba8b0bed73fb3633c86467352d74dea479c4330098d79316e15e1a2bf3b95a")
+  set(_itk_sha512 "ecf7e98564d8b9291df1d72a4169ec84e4dd5b5f605cb8d8d1ca6983a1c2b0abcbff5f9c468fe5cf59724903ce0d74f07840062961e2d5728dcd3268fc22f5f6")
```
Comment (cryos) [inline versions.cmake]: Was this built with Python 2.7 or 3.6 - the patch above suggests 2.7

Comment (mathturtle) [inline versions.cmake]: Unfortunately it looks like it was built with Python 2.  I'm rebuilding it now.  It was missing a few variables to make it use the python from the tomviz superbuild too.  I'm guessing it didn't get updated last time we built ITK and was out of date.

Comment (mathturtle) [inline versions.cmake]: Updated with Python3 version.

**Pull request merged**